### PR TITLE
Feature/create login register

### PR DESCRIPTION
**Pull request by RomanKuk**
now user is being added to db and returned to frontend
need to change firebaseConfig in frontend and FirebaseProjectName in appsetting.json
added sing in via github and google
important: now this works without user validation, so when you sing in, you just create a brand new user in db

### CODE REVIEW

File: frontend/src/app/core/services/authentication.service.ts [hunk 1]
```diff
@@ -1,50 +1,103 @@
...
-  public logout() {
-    this.user = undefined;
-    this.removeUserFromStorage();
+  doGoogleSignUp(credential: firebase.auth.UserCredential) {
+    const user = {} as User;
+    user.email = credential.user.email;
+    user.username = credential.user.displayName;
+    user.avatarUrl = credential.user.photoURL;
+    user.firstName = credential.additionalUserInfo.profile[`given_name`];
+    user.lastName = credential.additionalUserInfo.profile[`family_name`];
```
Comment (VoBilyk) [inline frontend/src/app/core/services/authentication.service.ts:56]: Better to write in the next style:
const user = {
    email: credential.user.email,
    username: credential.user.displayName,
    avatarUrl: credential.user.photoURL,
    firstName: credential.additionalUserInfo.profile[`given_name`],
    lastName: credential.additionalUserInfo.profile[`family_name`],
} as User;

File: frontend/src/app/core/services/authentication.service.ts [hunk 2]
```diff
@@ -1,50 +1,103 @@
...
+
+  doGithubSignUp(credential: firebase.auth.UserCredential) {
+    const user = {} as User;
+    user.email = credential.user.email;
+    user.username = credential.additionalUserInfo.username;
+    user.avatarUrl = credential.user.photoURL;
+    const name: string = credential.additionalUserInfo.profile[`name`];
+    if (name !== null)
+    {
+      const names: string[] = name.split(' ');
```
Comment (VoBilyk) [inline frontend/src/app/core/services/authentication.service.ts:73]: Try this: const [firstName, lastName = ''] = name.split(' ');

File: frontend/src/app/core/services/authentication.service.ts (hunk 1 above)
Comment (VoBilyk) [reply frontend/src/app/core/services/authentication.service.ts:56]: Do for github the same

**Comment by RomanKuk** [comment]
add users to db after login/register
add migrations: add data to social networks, add uid column ro userSN
important: need to fix: user who has the same email in github and google can login only via one of them
need to add validation to user properties

**Pull request merged**